src/NuGet.Core/NuGet.Configuration/Settings/SettingsLoadingContext.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+
+namespace NuGet.Configuration
+{
+    public class SettingsLoadingContext : IDisposable
+    {
+        private IList<SettingsFile> _settingsFiles = new List<SettingsFile>();
+        private readonly SemaphoreSlim _semaphore;
+        private bool _isDisposed;
+
+        public SettingsLoadingContext()
+        {
+            _semaphore = new SemaphoreSlim(1, 1);
+        }
+
+        internal SettingsFile GetOrCreateSettingsFile(string filePath)
+        {
+            if (filePath == null)
+            {
+                throw new ArgumentNullException(string.Format(CultureInfo.CurrentCulture, Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(filePath)));
+            }
+            _semaphore.Wait();
+            try
+            {
+                for (int i = 0; i < _settingsFiles.Count; i++)
+                {
+                    if (_settingsFiles[i].ConfigFilePath.Equals(filePath))
```

Comment:
Casing differences and non-normalized paths won't be an issue?

Reply:
yeah, I will fix that.
Don't remember why I didn't do it yesterday. Pretty sure the change is just sitting on a machine not pushed :)